### ISSUE TITLE
Issue with niigz2nii ? 

### ISSUE THREAD

**Issue by nicofarr** [individual-brain-charting/public_analysis_code#7]
Hi,

I am trying to run the preprocessing on the hcp1 protocol as a start. I have set the paths in the ini files, as well as changed the paths in the pipeline.py
- Running pipeline.py makes a few successful fsl topup (although many of them fail with a segfault / core dump) , but for example it does work for sub-01 ses-03.
- Next step, crashes when trying to NewSegment saying that it doesn't accept nii.gz file (comes from SPM) although the node niigz2nii seems to have been launched previously. 
[output.txt](https://github.com/hbp-brain-charting/public_analysis_code/files/5175316/output.txt)

I have attached the full output. Does this error ring a bell ? 

Thanks in advance for your help,

**Comment by bthirion** [comment]
OK, I don't expect an issue wit niigz2nii.
What happens when you try to launch pypreprocess examples directly ?
e.g. https://github.com/neurospin/pypreprocess/blob/master/examples/easy_start/nipype_preproc_spm_auditory.py

**Comment by nicofarr** [comment]
I was successful with launching [this one on haxby](https://github.com/neurospin/pypreprocess/blob/master/examples/easy_start/nipype_preproc_spm_haxby.py)

When trying the SPM auditory one, I get the error that nilearn.stats doesn't exist (which I assume is because I am using a pypi release of nilearn in which nistats wasn't yet merged).

**Comment by nicofarr** [comment]
Do I need to work with the development version of nilearn ?

**Comment by swaythe** [comment]
We recently updated the nistats functions with nilearn.stats functions. If you re-install pypreprocess you should not have the nilearn.stats issue.

**Comment by nicofarr** [comment]
Ok, now the example nipype_preproc_spm_auditory runs successfully. 
Note that I found another version issue along the way : Python > 3.6 won't work due to (presumably) [this issue](https://bugzilla.redhat.com/show_bug.cgi?id=1827745), and that seems easy to solve by not

This part of the discussion would actually belong in an issue in pypreprocess. 

I'll test the public_analys_code script later on and let you know. Thanks so far ! ;)

**Comment by bthirion** [comment]
Please open an issue in Pypreprocess.
Best,
B

**Comment by nicofarr** [comment]
@bthirion I will open a separate issue on pypreprocess repo.

Back on the current problem : I tested the public_analysis_code again (after having installed the nilearn version from swaythe master and reinstalled pypreprocess ) and I still encounter the same issue. 

Precisely : 

`Each element of the 'channel_files' trait of a NewSegmentInputSpec instance must be a pathlike object or string representing an existing file, but a value of '/media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat/sub-01_ses-00_T1w.nii.gz' <class 'str'> was specified.`

This seems linked to this [closed nipype issue](https://github.com/nipy/nipype/issues/2984)

Noticing that my environment had nipype 0.15.0, I downgraded to  0.14.0, now the error is different : 

`traits.trait_errors.TraitError: /media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat/sub-01_ses-00_T1w.nii.gz is not included in allowed types: .img, .hdr, .nii`

**Comment by bthirion** [comment]
Could be that `/media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat/sub-01_ses-00_T1w.nii.gz does not exist.`Is it the case ?

**Comment by nicofarr** [comment]
It does exist, it was actually copied by the pipeline to this location, the source data is in `/media/nfarrugi/datasets/ibc/`

**Comment by bthirion** [comment]
OK. Can you upgrade Nipype to more recent version then ?
Best,

**Comment by nicofarr** [comment]
Hi, sorry if the previous comment was unclear ; when using the more recent version (0.15.0) , I get this error:  

`Each element of the 'channel_files' trait of a NewSegmentInputSpec instance must be a pathlike object or string representing an existing file, but a value of '/media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat/sub-01_ses-00_T1w.nii.gz' <class 'str'> was specified.`

**Comment by nicofarr** [comment]
```
(base) nfarrugi:anat/ $ pwd                                           [8:41:13]
/media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat
(base) nfarrugi:anat/ $ ls                                            [8:42:01]
sub-01_ses-00_acq-spc_T2w.nii  sub-01_ses-00_T1w.nii
sub-01_ses-00_acq-tse_T2w.nii  sub-01_ses-00_T1w.nii.gz
sub-01_ses-00_FLAIR.nii

```
Still suspecting that something is going wrong between niigz2nii and the subsequent step, because the error returned is : 

`traits.trait_errors.TraitError: /media/nfarrugi/datasets/ibc/derivatives/sub-01/ses-00/anat/sub-01_ses-00_T1w.nii.gz is not included in allowed types: .hdr, .img, .nii`

as if the connection between the niigz2nii and the NewSegment module was not done.

**Comment by bthirion** [comment]
> Hi, sorry if the previous comment was unclear ; when using the more recent version (0.15.0) , I get this error:

But there are some more recent versions of nipype.

**Comment by nicofarr** [comment]
ok I'll try building nipype from source then

**Comment by nicofarr** [comment]
Just built from the master branch on github  https://github.com/nipy/nipype and still getting the same error.

**Comment by bthirion** [comment]
Argh. @swaythe can you provide a setup on which all this runs ?
Best

**Comment by swaythe** [comment]
Yeah, I'll take a look.

**Comment by swaythe** [comment]
@nicofarr In the ini file, I noticed that the anatomical is specified as a .nii file instead of a .nii.gz file (line 83 of the hcp1 ini file). Could you change that to nii.gz and run the script again and tell me if/where it fails?

**Comment by swaythe** [comment]
@nicofarr I managed to replicate your error and think I know the issue. It'll probably be a few days before it gets fixed though. In the mean time, if you want the preprocessed HCP files please let us know.

**Comment by bthirion** [comment]
You mean: "the IBC files acquired under HCP task". Indeed, I can put them  on an ftp server.

**Comment by swaythe** [comment]
Yes, those. Thanks for the clarification @bthirion . :~)

**Comment by nicofarr** [comment]
Hi ! Sorry for the delay, so as you might have guessed, there was no immediate rush. :) I anticipated it may take a while. We are doing a first analysis using the maps on neurovault, and we will most likely need the BOLD by the end of October.

**Comment by swaythe** [comment]
Hi @nicofarr , sorry for the delay! From my tests, it appears that the error you were getting was due to a missing flag in pypreprocess that is activated when a single subject's data is analyzed. I have fixed it and filed a PR, but the changes are currently only available on my fork of pypreprocess (https://github.com/swaythe/pypreprocess). We will hopefully fix that next week. There were also a couple of changes in the analysis scripts relating to nistats being replaced by nilearn.glm, which should now be available to you on the repo. Let me know if you encounter other errors.
